lib/oxigraph/src/sparql/dataset.rs

```diff
@@ -1,18 +1,235 @@
+use crate::model::NamedNodeRef;
 use crate::sparql::QueryDataset;
 use crate::storage::numeric_encoder::{
     insert_term, Decoder, EncodedTerm, EncodedTriple, StrHash, StrHashHasher, StrLookup,
 };
 use crate::storage::{CorruptionError, StorageError, StorageReader};
+use hdt::Hdt;
 use oxrdf::Term;
 use oxsdatatypes::Boolean;
 use spareval::{ExpressionTerm, ExpressionTriple, InternalQuad, QueryableDataset};
 use std::cell::RefCell;
 use std::collections::hash_map::Entry;
 use std::collections::HashMap;
 use std::hash::BuildHasherDefault;
+use std::io::{Error, ErrorKind};
 use std::iter::empty;
+use std::str::FromStr;
 use std::sync::Arc;
 
+pub struct HDTDataset {
```

Comment:
If you add `#[derive(Clone)]` here you can then also add it on `#[derive(Clone)]` and remove the custom implementation